app/generic/definitions.py

```diff
@@ -0,0 +1,390 @@
+    # done inside the endpoint implementation.
+    aries_controller: AcaPyClient = Depends(agent_role([Role.YOMA, Role.ECOSYSTEM])),
+    auth: AcaPyAuthVerified = Depends(acapy_auth_verified),
+):
+    """
+    Create a credential definition.
+
+    Parameters:
+    -----------
+    credential_definition: CreateCredentialDefinition
+        Payload for creating a credential definition.
+
+    Returns:
+    --------
+    Credential Definition
+    """
+
+    # Assert the agent has a public did
+    public_did = await acapy_wallet.assert_public_did(aries_controller)
+
+    # Make sure we are allowed to issue this schema according to trust registry rules
+    await trust_registry.assert_valid_issuer(
+        public_did, credential_definition.schema_id
+    )
+
+    wait_for_event, stop_listener = await start_listener(
+        topic="endorsements", wallet_id=auth.wallet_id
+    )
+
+    result = await aries_controller.credential_definition.publish_cred_def(
+        body=CredentialDefinitionSendRequest(
+            schema_id=credential_definition.schema_id,
+            # Revocation not supported yet
+            support_revocation=False,
+            tag=credential_definition.tag,
+        )
+    )
+
+    if isinstance(result, TxnOrCredentialDefinitionSendResult):
+        try:
+            # Wait for transaction to be acknowledged and written to the ledger
+            await wait_for_event(
+                filter_map={
+                    "state": "transaction-acked",
+                    "transaction_id": result.txn.transaction_id,
+                }
+            )
+        except asyncio.TimeoutError:
+            raise CloudApiException(
+                "Timeout waiting for endorser to accept the endorsement request"
+            )
+
+        try:
+            transaction = await aries_controller.endorse_transaction.get_transaction(
+                tran_id=result.txn.transaction_id
+            )
+
+            # Based on
+            # https://github.com/bcgov/traction/blob/6c86d35f3e8b8ca0b88a198876155ba820fb34ea/services/traction/api/services/SchemaWorkflow.py#L276-L280
+            signatures = transaction.signature_response[0]["signature"]
+            endorser_public_did = list(signatures.keys())[0]
+            signature = json.loads(signatures[endorser_public_did])
+
+            public_did = signature["identifier"]
+            sig_type = signature["operation"]["signature_type"]
+            schema_ref = signature["operation"]["ref"]
+            tag = signature["operation"]["tag"]
+            credential_definition_id = f"{public_did}:3:{sig_type}:{schema_ref}:{tag}"
+        except Exception as e:
+            raise CloudApiException(
+                "Unable to construct credential definition id from signature response"
+            ) from e
+
+        # FIXME: ACA-Py 0.7.3 has no way to associate the credential definition id with a transaction record
```

Comment:
Are we leaving this out for now then?

Reply:
Ah no I found a workaround for this (see `credential_definition_id = f"{public_did}:3:{sig_type}:{schema_ref}:{tag}"` above). Will remove 👍